### PR TITLE
Fixes codespell

### PR DESCRIPTION
**Pull request by erak**
Fixes:
```
./libevmasm/RuleList.h:51: hist  ==> heist, his
./libevmasm/RuleList.h:51: hist  ==> heist, his
./docs/yul.rst:247: varn  ==> warn
./docs/yul.rst:248: varn  ==> warn
./docs/yul.rst:249: varn  ==> warn
./docs/yul.rst:252: varn  ==> warn
./test/compilationTests/corion/provider.sol:152: datas  ==> data
./test/compilationTests/corion/provider.sol:275: datas  ==> data
./test/compilationTests/corion/provider.sol:326: datas  ==> data
```

### CODE REVIEW

File: scripts/codespell_whitelist.txt [hunk 1]
```diff
@@ -9,3 +9,5 @@ ba
 fo
 compilability
 errorstring
+varn
```
Comment (chriseth) [inline scripts/codespell_whitelist.txt]: Maybe use `var_n` instead?

**Comment by ekpyron** [comment]
By the way errorstring and assignend are both in the whitelist, but we only have errorString and assignEnd in the repo - is it possible to make codespell gracefully deal with camelCase? Unrelated to this PR, though. [EDIT: https://github.com/codespell-project/codespell/issues/196 and https://github.com/codespell-project/codespell/issues/314 - doesn't look like it currently]

File: scripts/codespell_whitelist.txt (hunk 1 above)
Comment (erak) [reply scripts/codespell_whitelist.txt]: Yeah, then we would need to apply that everywhere like `arg1, argn`, `l1, ..., ln`.

Comment (chriseth) [reply scripts/codespell_whitelist.txt]: there are only 2 places, right?

**Review by chriseth — APPROVED** [review]

**Pull request merged**